### ISSUE TITLE
Getting repeating high loss and 0 accuracy

### ISSUE THREAD

**Issue by mkmitchell** [jeffwen/road_building_extraction#2]
Thanks for posting your code.  It's helping me wrap my head around neural networks and pytorch.

My task is satellite image classification.  I have a 10 band multi spectral image that I've padded and gridded into 128x128 images.  The last band is the class (0-6).  I modified your data loader (really nice job by the way) so I have a 9x128x128 tensor for inputs and a 128x128 tensor for labels.  I modified the model to take in 9 bands
```
        # encoding
        self.conv1 = encoding_block(9, 32)
```
My training and validation tensors appear to be correct when training the model.  I checked this by printing out a small section from both sat and map in the train method.

My small test dataset has 63 images for training, 7 for validation, and 16 for testing.  So far I'm able to set my batch size to 32 (8gb GTX1070) which means I only have two iterations per epoch.

I'm happy that it's running but every epoch produces the same loss and accuracy.

```
Epoch 71/99
----------
training: 100%|██████████████████████████████████████████████| 2/2 [00:02<00:00,  1.74s/it]
Training Loss: 1046.7444 Acc: 0.0000

validation: 100%|████████████████████████████████████████████| 3/3 [00:01<00:00,  1.57it/s]
Validation Loss: 693.5661 Acc: 0.0000

Current elapsed time 5m 33s
```

[The notebook I'm working from](https://github.com/mkmitchell/PlayWithPyTorch/blob/master/Modelrun.ipynb)

Any advice would be greatly appreciated.
Thank!

**Comment by mkmitchell** [comment]
I would imagine it's the line
`outputs = torch.nn.functional.sigmoid(outputs)`

I believe sigmoid returns 0-1.  I'll read into functions.

**Comment by sunshinezhihuo** [comment]
@mkmitchell Hi. Do you solve your problem now?

**Comment by wangzq0629** [comment]
hi， it is no diffcult，I do it，thank you！



wangwq@mapcore.com.cn
 
From: Qi
Date: 2019-03-29 09:46
To: jeffwen/road_building_extraction
CC: Subscribed
Subject: Re: [jeffwen/road_building_extraction] Getting repeating high loss and 0 accuracy (#2)
@mkmitchell Hi. Do you solve your problem now?
―
